### ISSUE TITLE
valid-types provides error with using import from other file

### ISSUE THREAD

**Issue by krutoo** [gajus/eslint-plugin-jsdoc#343]
Hi. Please help me resolve problem with "jsdoc/valid-types" rule. Import of types provides error:

![image](https://user-images.githubusercontent.com/13140130/61359121-2f612580-a895-11e9-8bda-8f80fa53fbec.png)

error text: "JSDoc syntax error"

`eslint-plugin-jsdoc` version: 15.5.2
`eslint` version: 6.0.0

eslint rule configuration:
```json5
{
  // ...
  rules: {
    // ...
    "jsdoc/valid-types": "error"
  }
}
```

**Comment by brettz9** [comment]
Indeed, support for the TypeScript `import()` syntax has been [added](https://github.com/jsdoctypeparser/jsdoctypeparser/issues/50) in the underlying jsdoctypeparser and testing this code in our testing environment (and checking the type in the [jsdoctypeparser demo](https://jsdoctypeparser.github.io/)) reports no issues:

```js
/**
       * Component.
       * @param {import('../properties-list/properties-list.jsx').PropertyType} prop Property.
       */
       const DownloadList = ({ files }) => {};
```

Please be sure to restart your IDE after you have installed updates to `eslint-plugin-jsdoc`. With the Atom linting package at least, it apparently caches the files.

**Comment by brettz9** [comment]
I expect this is a stale files issue, so closing, but feel free to comment further and we can reopen if we can replicate.

**Comment by krutoo** [comment]
i have problem with linting in terminal. It provides next error:
![image](https://user-images.githubusercontent.com/13140130/61429999-def3d180-a941-11e9-9851-9c7c12ae33ad.png)

I don't use cache in ESLint. I restart my IDE (VS Code, PHPStorm) but i think that is independent things.

**Comment by brettz9** [comment]
Ok, and you are sure you have updated to the latest eslint-plugin-jsdoc, along with its dependencies? Your `yarn.lock` file should be showing version "5.0.1" for the dependency "jsdoctypeparser" because earlier versions did not properly support `import()`.

If you have confirmed this, can you reduce your test case to a single erring block, and reduce your `.eslintrc` to the minimum set of rules?

**Comment by krutoo** [comment]
@brettz9 here is screenshot of `yarn.lock` file:
![image](https://user-images.githubusercontent.com/13140130/61433733-87f3f980-a94d-11e9-9873-1d3870629903.png)

I already have only one erring block now. 

I understood that rule `valid-jsdoc` provides this error. After remove this rule i have no errors. 

But this rule is necessary for my project. What can i do?

**Comment by brettz9** [comment]
Ok, good to see you have the dependency, but by a single erring block, I mean you have no other code in the file that triggers the error. And to be safe, try removing all other eslint rules and configs/plugins and paste your exact code and config that still reproduces the issue.

**Comment by krutoo** [comment]
That config (with old "valid-jsdoc" rule) provides error:
```json
{
  "rules": {
    "valid-jsdoc": [
      "error",
      {
        "prefer": {
          "arg": "param",
          "argument": "param",
          "class": "constructor",
          "virtual": "abstract",
          "returns": "return"
        },
        "preferType": {
          "any": "*",
          "Null": "null",
          "Undefined": "undefined",
          "Boolean": "boolean",
          "Number": "number",
          "int": "number",
          "integer": "number",
          "float": "number",
          "String": "string",
          "Symbol": "symbol",
          "object": "Object",
          "array": "Array",
          "function": "Function"
        },
        "requireReturn": false,
        "requireReturnType": true,
        "requireParamType": true,
        "matchDescription": ".+",
        "requireParamDescription": true,
        "requireReturnDescription": true
      }
    ],
  }
}
```

In this code:
```javascript
/**
 * Component.
 * @param {import('../properties-list/properties-list.jsx').PropertyType} prop Property.
 */
const DownloadList = ({ files }) => {/* ... */};
```

But i just remove "valid-jsdoc" rule and replace it on rules from this plugin. And it works without errors.
I non't now it issue is needed now, because "valid-jsdoc" is deprecated rule. Maybe it can be closed.

**Comment by nemoDreamer** [comment]
This is still happening on `eslint-plugin-jsdoc@35.4.1` in VS Code:

![image](https://user-images.githubusercontent.com/9303/124364377-1c28fb00-dc0f-11eb-8da5-807acde7e420.png)

But VS Code does correctly find the type this way:

![image](https://user-images.githubusercontent.com/9303/124364499-eb959100-dc0f-11eb-8c5f-b39521bcfeba.png)

---

I'm using
```json
{
    extends: [
        "airbnb-base",
        "plugin:jsdoc/recommended",
        "plugin:prettier/recommended",
    ],
    rules: [
        // deprecated rule conflicting with `eslint-plugin-jsdoc`?
        "valid-jsdoc": "off",
        "require-jsdoc": "off",
    ],
}
```

... but also happens with _only_ `plugin:jsdoc/recommended`...

**Comment by brettz9** [comment]
To avoid errors, you need to set `settings.jsdoc.mode` to "typescript" or to include `@typescript/eslint-parser` as your parser.

**Comment by nemoDreamer** [comment]
Thank you, @brettz9, that was the missing piece!
